src/Browser/Session.php

```diff
@@ -95,26 +96,38 @@ public function expectException($expectedException, ?string $expectedMessage = n
 
     public function source(): string
     {
-        $ret = "<!--\n";
+        $ret = '';
 
         try {
-            $ret .= "URL: {$this->getCurrentUrl()} ({$this->getStatusCode()})\n\n";
+            $isTextContent = str_contains((string) $this->getResponseHeader('content-type'), 'text')
+                || str_contains((string) $this->getResponseHeader('content-type'), 'json');
+        } catch (UnsupportedDriverActionException) {
+            $isTextContent = true;
+        }
```

Comment:
please, could you comment out why this exception leads to `$isTextContent = true;`?

Reply:
The exception is thrown by `Behat\Mink\Exception\UnsupportedDriverActionException: Response headers are not available from Zenstruck\Browser\Session\Driver\PantherDriver`, so to avoid BC break, I prefer set it to true, as it is actually

Reply:
what happen if we use Panther and try to get a zip, then?

Reply:
The zip file will be corrupted. 
The only way to avoid it using Panther is to use a `$raw` parameter

Reply:
PantherDriver indirectly extends https://github.com/minkphp/Mink/blob/master/src/Driver/CoreDriver.php#L287 (through out https://github.com/welcoMattic/browser/blob/2cd69e9be35a50617a7a8ac9620c2f6e5d5eb35f/src/Browser/Session/Driver.php#L24), so I guess we don't have any way to retrieve response header from Panther? right @kbond ?

Reply:
Correct, we can't get the status code/headers from Panther.